projects/igniteui-angular/src/lib/services/overlay/position/auto-elastic-position-strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { ConnectedFit, Size } from './../utilities';
+import { AutoPositionStrategy } from './auto-position-strategy';
+
+/**
+ * TODO
+ */
+export class AutoElasticPositionStrategy extends AutoPositionStrategy {
+    /** @inheritdoc */
+    position(contentElement: HTMLElement, size: Size, document?: Document, initialCall?: boolean): void {
+        this._forceFit = true;
+        super.position(contentElement, size, document, initialCall);
+    }
+
+    /** @inheritdoc */
+    protected fitInViewport(element: HTMLElement, connectedFit: ConnectedFit) {
+        const heightOverflow = connectedFit.contentElementRect.height - connectedFit.viewPortRect.height;
+        if (heightOverflow > 0) {
+            element.style.width = 'auto';
```

Comment:
do not set the width to `auto`. Instead add logic to check if the content goes out in horizontal direction and set element's `width` as done for `height`

Reply:
As we discussed this strategy will be only used for the excel style filtering, so leaving the width to 'auto'.